files/common/usr/bin/get-appliance-version

```diff
@@ -18,6 +18,76 @@
 set -o errexit
 set -o pipefail
 
-zfs get -Hpo value \
+function usage() {
+	echo "Usage: $(basename "$0") [-dMmps]"
+	echo
+	echo "Display the version information for the appliance. By default,"
+	echo "the full version (major, minor, and patch portions) are displayed."
+	echo "The following arguments may be used to refine the version"
+	echo "information:"
+	echo "-d, --debug -- provide the entire version including the jenkins"
+	echo "               information, if it exists."
+	echo "-M, --major -- displays the major version"
+	echo "-m, --minor -- displays the major and minor version"
+	echo "-p, --patch -- displays the major, minor, and patch version"
+	echo "-s, --snapshot -- displays the major, minor, patch, and " \
+		"snapshot information"
+	exit 2
+}
+
+options=$(getopt -l "major,minor,patch,snapshot,debug" -o "Mmpsd" -- "$@")
+eval set -- "$options"
+
+output=$(zfs get -Hpo value \
 	"com.delphix:current-version" \
-	"$(dirname "$(zfs list -Hpo name /)")" | tr -d '\n'
+	"$(dirname "$(zfs list -Hpo name /)")")
+
+opt_major=false
+opt_minor=false
+opt_patch=false
+opt_debug=false
+opt_snapshot=false
+
+if [[ $# -gt 2 ]]; then
+	echo "Error: Only one option may be selected"
+	usage
+fi
+
+while [[ $# -gt 0 ]]; do
+	case "$1" in
+	-d | --debug)
+		opt_debug=true
+		shift
+		;;
+	-M | --major)
+		opt_major=true
+		shift
+		;;
+	-m | --minor)
+		opt_minor=true
+		shift
+		;;
+	-p | --patch)
+		opt_patch=true
+		shift
+		;;
+	-s | --snapshot)
+		opt_snapshot=true
+		shift
+		;;
+	--)
+		shift
+		break
+		;;
+	esac
+done
+
+$opt_debug && echo "$output" | tr -d '\n'
+$opt_major && echo "$output" | cut -d'.' -f1,2 | tr -d '\n'
+$opt_minor && echo "$output" | cut -d'.' -f1-3 | tr -d '\n'
+$opt_patch && echo "$output" | cut -d'-' -f1 | cut -d'.' -f1-4 | tr -d '\n'
+$opt_snapshot && echo "$output" | cut -d'+' -f1
+
+if ! $opt_major && ! $opt_minor && ! $opt_patch && ! $opt_debug; then
+	echo "$output" | cut -d'-' -f1 | tr -d '\n'
```

Comment:
Why don't we want the default behaviour to print out the full version?

e.g.
```
$ ./get-appliance-version
6.1.0.0
```

The actual "appliance version" includes all the additional information, which is what's displayed in the UI and tracked in MDS. This enables us to do things like trunk to trunk upgrades, since the actual version is different. This means, any app-gate consumers that need the full version will use the `--debug` option, which feels a little awkward to me.

I don't feel strongly; just wanted to share my thoughts as I'm reading over this change...

Reply:
> Why don't we want the default behaviour to print out the full version?

Sorry, just to be clear, I mean why don't we want to print the "debug" version by default..

Reply:
Sure, I wasn't aware that we stored this.

Reply:
![image](https://user-images.githubusercontent.com/658081/78174551-4b4e2c80-740e-11ea-8c67-9e072d396262.png)

Yes, currently the app-gate uses all of the version information, including the snapshot and build information. Additionally, the `VERSION` of an upgrade image, also includes all of the snapshot and build information in it.

Reply:
With that said, I think we could probably make do with the default only including the snapshot information, and not the build/jenkins information.

I think the main constraint is the version is different between builds, so that we can easily support trunk to trunk upgrades. I believe the snapshot information alone satisfies that, so I'd be good with using `-s` as the default mode.

Reply:
I'm assuming the snapshot and jenkins info only exists on non-released builds. Is that true? I think leaving the old semantics is the safest thing since that's what we've been using. Anybody wanting some portion of the version should supply the necessary option.

Reply:
> Is that true?

Yes, as far as I understand, that's true.

> I think leaving the old semantics is the safest thing since that's what we've been using.

Sounds good to me.